services/data/src/links/RestAPILink/fetchData.ts

```diff
@@ -10,33 +10,51 @@ export const parseContentType = (contentType: string | null) => {
 }
 
 export const parseStatus = async (response: Response) => {
-    if (
+    const accessError =
         response.status === 401 ||
         response.status === 403 ||
         response.status === 409
-    ) {
-        const message = await response
-            .json()
-            .then(body => {
-                return body.message
-            })
-            .catch(() => {
-                return response.status === 401 ? 'Unauthorized' : 'Forbidden'
-            })
+
+    if (accessError) {
+        let message
+        let details = {} as any
```

Comment:
I think I saw you mention this in a comment, sorry I missed it - the issue here I assume is that `details` is assigned the type of an **empty** object with `let details = {}`, and so typescript complains about dereferencing `details.message`, is that right?

If so, I think the right thing to do is to change the FetchErrorDetails type to be what we actually expect it to be - an Object with an optional `message` prop.  So that would be 

Once the type is changed we can actually say `let details: FetchErrorDetails = {}` 

I tested this quickly [here](https://www.typescriptlang.org/play/?ssl=1&ssc=47&pln=1&pc=1#code/C4TwDgpgBAYhwGMAWBRATmg9mgIvAhgJYA2AzlALxQDeUAthKafgObQD8AXFKcGoQDsWUAL4AoYvCgATAiVLc4iVBmx5gRMpRrixsjfO20GTVhG4ByAB4gAXhdFR85fAJBiEmAaUySAdMSYLAAU+pqkfibMbACUQA), seems to work and has the added benefit of being extra explicit about what we expect to be in the `details` object.

Reply:
Ah ok, thanks. I've updated it to match your suggestions 👍